ios/FluentUI/Core/SwiftUI+ViewPresentation.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// Modifier that presents a SwiftUI View as an additional layer on top of the content View.
+struct PresentationModifier<PresentedView: View>: ViewModifier {
+    func body(content: Content) -> some View {
+        ZStack {
+            content
+
+            // Backdrop for the presented view which overlays the
+            // content where it's being presented from.
+            Rectangle()
+            .frame(maxWidth: .infinity,
+                   maxHeight: .infinity,
+                   alignment: .center)
+            .foregroundColor(isPresented ? backgroundColor : Color.clear)
+            .allowsHitTesting(isPresented && isBlocking)
+
+            presentedView
+        }
+        .frame(maxWidth: .infinity,
+               maxHeight: .infinity,
+               alignment: .center)
+    }
+
+    var hasDimmedBackground: Bool
+    var dimmedBackgroundColor: Color
+    var dimmedBackgroundColroOpacity: Double
+    var isBlocking: Bool
+    @Binding var isPresented: Bool
+    @ViewBuilder var presentedView: PresentedView
+
+    var backgroundColor: Color {
+        if hasDimmedBackground {
+            return dimmedBackgroundColor.opacity(dimmedBackgroundColroOpacity)
+        } else if isBlocking {
+            // A clear color won't allow the background
+            // view to block the underlying view
+            return Color.black.opacity(Double.ulpOfOne)
+        }
+
+        return Color.clear
+    }
+}
+
+extension View {
+
+    /// Presents a View as a layer on top of this View.
+    /// - Parameters:
+    ///   - isPresented: Binding boolean value that determines whehter the `presentedView` is presented on top of this view..
```

Comment:
nit: missing the dimmed background parameters here

Reply:
Thanks for catching!